arangod/Cluster/ServerState.cpp

```diff
@@ -676,15 +680,18 @@ bool ServerState::checkEngineEquality(AgencyComm& comm) {
       return true;  // do not do anything harsh here
     }
 
-    for (VPackObjectIterator::ObjectPair pair : VPackObjectIterator(servers)) {
-      if (pair.value.isObject()) {
-        std::string_view const engineName =
-            _server.getFeature<EngineSelectorFeature>().engineName();
+    std::string_view engineName =
+        _server.getFeature<EngineSelectorFeature>().engineName();
 
-        VPackSlice engineStr = pair.value.get("engine");
-        if (engineStr.isString() && !engineStr.isEqualString(engineName)) {
-          return false;
-        }
+    for (auto pair : VPackObjectIterator(servers)) {
+      if (!pair.value.isObject()) {
+        continue;
+      }
+
+      VPackSlice engine = pair.value.get("engine");
+      if (engine.isString() && engine.stringView() != engineName) {
```

Comment:
Not really. But it is just old code that I lightly modified to get rid of the call to `isEqualString()`. I think we used `isEqualString()` long ago to avoid copying the contained string, but as we use stringView() nowadays, we should use whatever std::string_view offers.
The old code was logically equivalent:
```
        VPackSlice engineStr = pair.value.get("engine");
        if (engineStr.isString() && !engineStr.isEqualString(engineName)) {
          return false;
        }